File: DataBindingListView/app/src/main/java/com/example/databindinglistview/MainActivity.java
package com.example.databindinglistview;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.database.DataSetObserver;
import android.os.Bundle;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListAdapter;
import android.widget.ListView;

import com.example.databindinglistview.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;  // used to create binding from xml..

    String name[] = {"Java", "Android", "Php", "Js"};

    int images[] = {R.drawable.java, R.drawable.android, R.drawable.php, R.drawable.js};


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        ListViewAdapter listViewAdapter = new ListViewAdapter(MainActivity.this, name, images);

        binding.lvName.setAdapter((ListAdapter) listViewAdapter);

    }
}

File: DataBindingExample/app/src/main/java/com/example/databindingexample/MainActivity.kt
package com.example.databindingexample

import android.database.DatabaseUtils
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.databinding.DataBindingUtil
import androidx.databinding.DataBindingUtil.setContentView

import com.example.databindingexample.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding =DataBindingUtil.setContentView(this,R.layout.activity_main)
        val quote = Quotes("Life is an art","Sohil Upadhyay")
        binding.quotes = quote
    }
}